src/Collection/CollectionTrait.php

```diff
@@ -502,4 +507,19 @@ public function unfold(callable $transformer = null)
             )
         );
     }
+
+    /**
+     * Returns the closest nested iterator that can be safely traversed without
+     * losing any possible transformations.
+     *
+     * @return \Iterator
+     */
+    protected function _unwrap()
+    {
+        $iterator = $this;
+        while (get_class($iterator) === 'Cake\Collection\Collection') {
```

Comment:
@lorenzo To allow people to extend the cake collection class and have the benefit of this unwrap shouldn't use `instanceof` here? It is actually faster than `get_class()` too.